### DISCUSSION TITLE
Android studio error: "Cannot resolve symbol 'XX'"

### DISCUSSION THREAD

**Discussion by inmess** [callstack/react-native-builder-bob#403] [Q&A] [answered]
I'm new to native development, using 'npx create-react-native-library@latest' to initialized a module. However, I got stuck in the very first step...

I really have no clue whether it is about my sys environment (_something like jdk version, node version or gradle_) or a bug of this lib (_which I don't think so_)... 

BTW gradle really did mess me up a lot before this...

Anyway, literally my project was not changed at all, error occurred right after I init it and open `android/` folder in Android Studio.

System: Macos with M1 chip;
JDK version: 17 (I do have two different versions on my computer, jdk-11 is the one default in PATH, 17 is the one installed by AS);
react-native version: 0.72.1;
node version: 16.19.0

![image](https://github.com/callstack/react-native-builder-bob/assets/48652450/e8718c47-3e09-4777-9a2b-036bce6fc2e3)

![image](https://github.com/callstack/react-native-builder-bob/assets/48652450/c882203c-5694-404c-a996-4dcbd7aae55a)

**Comment by atlj** [comment, marked as answer]
Hey, could you try opening `example/android` folder instead of the root level `/android` folder? This seems like to be the issue. Once you do this, you can switch to the Android tab and select your project to develop. Please make sure to take a look at the `CONTRIBUTING.md` file that is generated with your project. Line 58 should tell:

```
To edit the Java or Kotlin files, open `example/android` in Android studio and find the source files at `react-native-awesome-library` under `Android`.
```

**Reply by inmess** [reply]
Appreciate! Sorry for my ignorance...
I still wanna ask a bit more, so I will need to copy my change from `example/android/` to `android/` after development?

**Reply by atlj** [reply]
No problem 😅 and no you don't have to copy anything. All your changes should be reflected to the source files under the root level `/android` folder. You can verify it by checking the git changes.

**Reply by CyberCyclone** [reply]
I don't get how this answers the question. `example/android` is a different directory to `android/` and aren't even nested under each other. So how can they see each others project files? `example/android` contains the module files while `android/` only contains the `MainActivity` and `MainApplication` files.

**Reply by satya164** [reply]
> how can they see each others project files

https://callstack.github.io/react-native-builder-bob/faq#how-does-the-library-get-linked-to-the-example-app-in-the-generated-project

**Reply by atlj** [reply]
@CyberCyclone The example/android is also a gradle project. React Native links your android module to the example app so your module is added to the app as a gradle dependency. When you open your app in Android Studio, go to the gradle dependencies, find your lib, and edit it; the changes you make are within your fs. So the answer is Android Studio + React Native + Gradle handling those stuff for you.

**Reply by CyberCyclone** [reply]
Thanks @satya164 and @atlj, that makes sense. Thanks for the explanation.
